src/swsssdk/configdb.py

```diff
@@ -187,11 +190,16 @@ def set_entry(self, table, key, data):
             client.delete(_hash)
         else:
             original = self.get_entry(table, key)
-            client.hmset(_hash, self.__typed_to_raw(data))
-            for k in [ k for k in original.keys() if k not in data.keys() ]:
+            raw, _ = self.__typed_to_raw(data)
+            if len(raw):
+                client.hmset(_hash, raw)
+
+            for k in original.keys():
                 if type(original[k]) == list:
                     k = k + '@'
-                client.hdel(_hash, self.serialize_key(k))
+                if k not in raw.keys():
```

Comment:
Guess, you have to compare before adding "@". Did you test this code against list becoming none.
You might need the original line 191, with s/data/raw